### PR TITLE
Start DaemonSetJob implementation with API resource definition

### PR DESCRIPTION
**Pull request by clamoriniere1A**
This PR contains the definition of the DaemonSetJob API resource

### CODE REVIEW

**Comment by sdminonne** [comment]
@clamoriniere1A should I review this? Or do you want me to wait for you to remove 'WIP'?

**Comment by clamoriniere1A** [comment]
> @clamoriniere1A should I review this? Or do you want me to wait for you to remove 'WIP'?

You can start reviewing it. This PR is only the skeleton of the new controller

**Comment by sdminonne** [comment]
Started. Looks good but I need to run through all code, though...

**Comment by sdminonne** [comment]
@clamoriniere1A sorry to  ask this: may you split into three commits:

- api
- generated code
- controller stub
thanks

File: pkg/controller/daemonsetjob.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+	key, err := cache.MetaNamespaceKeyFunc(daemonsetjob)
+	if err != nil {
+		glog.Errorf("DaemonsetJobController:enqueue: couldn't get key for DaemonsetJob   %s/%s", daemonsetjob.Namespace, daemonsetjob.Name)
+		return
+	}
+	d.queue.Add(key)
+}
+
+// get workflow by key method
+func (d *DaemonsetJobController) getWorkflowByKey(key string) (*dapi.DaemonsetJob, error) {
```
Comment (EricMountain-1A) [inline pkg/controller/daemonsetjob.go]: Is it intentional to have a Workflow related method in the DaemonSetJob code?  So far it looked like we could upstream this if folks were interested after having proved it here, however this snippet would make that harder I suppose.

**Comment by clamoriniere1A** [comment]
> @clamoriniere1A sorry to ask this: may you split into three commits:
> 
> * api
> * generated code
> * controller stub
>   thanks

PR updated with 3 commits

File: pkg/controller/daemonsetjob.go (hunk 1 above)
Comment (clamoriniere1A) [reply pkg/controller/daemonsetjob.go]: done

**Comment by sdminonne** [comment]
LGTM. Thanks.
Merging since @EricMountain-1A comment is not on `types.go`.

**Pull request merged**